modules/core/src/test/scala/com/azavea/stac4s/Generators.scala

```diff
@@ -36,6 +38,25 @@ object Generators {
 
   private def instantGen: Gen[Instant] = arbitrary[Int] map { x => Instant.now.plusMillis(x.toLong) }
 
+  private def collectionExtensionFieldsGen: Gen[JsonObject] = Gen.oneOf(
+    Gen.const(().asJsonObject),
+    Gen
+      .mapOf(
+        (nonEmptyStringGen, stacCollectionAssetGen).tupled
+      )
+      .map(AssetCollectionExtension.apply)
+      .map(_.asJsonObject)
+  )
+
+  private def linkExtensionFields: Gen[JsonObject] = Gen.oneOf(
+    Gen.const(().asJsonObject),
+    Gen
+      .nonEmptyListOf(nonEmptyStringGen)
+      .map(NonEmptyList.fromListUnsafe)
+      .map(LabelLinkExtension.apply)
+      .map(_.asJsonObject)
+  )
```

Comment:
by adding these generators, we can ensure that with and without the extension fields, we can still roundtrip